x/events/events.go

```diff
@@ -356,3 +371,13 @@ func NewWebhookFailed(ctx context.Context, err error) (string, trace.EventOption
 			)...,
 		)
 }
+
+func reasonForError(err error) string {
+	if ve := new(schema.ValidationError); errors.As(err, &ve) {
+		return ve.Message
+	}
+	if r := *new(herodot.ReasonCarrier); errors.As(err, &r) {
+		return r.Reason()
+	}
+	return fmt.Sprintf("reason could not be determined for error: %T", err)
+}
```

Comment:
People will ask about "reason IDs" to find e.g. all failed logins for reason X over time. Don't our validation errors have some type of ID or error code that we could emit?

Reply:
I want to avoid people relying on these errors and then we change the casing or error message and we get complaints about backwards incompatible changes.

Reply:
Yes please

Reply:
We can also set some default like "contact_support" or something for the reason id to make it clear that you should request a reason id if it doesn't exist (opposed to reading the reason string field)